java/statestore/src/main/java/sleeper/statestore/s3/S3FileInfo.java

```diff
@@ -18,73 +18,113 @@
 
 import sleeper.core.statestore.FileInfo;
 
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class S3FileInfo {
-    public enum FileStatus {
-        ACTIVE, READY_FOR_GARBAGE_COLLECTION
-    }
 
-    private final FileInfo fileInfo;
-    private final FileStatus status;
+    private final String filename;
+    private final List<FileInfo> internalReferences;
+    private final int externalReferences;
```

Comment:
To distinguish the types of these 2 reference variables, maybe `externalReferences` could be renamed to `externalReferenceCount`?

Reply:
Done, thanks :)